src/dash-table/components/ControlledTable/TooltipCursorDecorator.tsx

```diff
@@ -0,0 +1,201 @@
+
+        const realIndex = rowIndex + virtualized.offset.rows;
+
+        const legacyTooltip = tooltips &&
+            tooltips[columnId] &&
+            (
+                tooltips[columnId].length > realIndex ?
+                    tooltips[columnId][realIndex] :
+                    null
+            );
+
+        const staticTooltip = column_static_tooltip[columnId];
+
+        const conditionalTooltips = R.filter(tt => {
+            return !tt.if ||
+                (
+                    (
+                        tt.if.column_id === undefined ||
+                        tt.if.column_id === columnId
+                    ) &&
+                    (
+                        tt.if.row_index === undefined ||
+                        tt.if.row_index === realIndex ||
+                        (tt.if.row_index === 'odd' && realIndex % 2 === 1) ||
+                        (tt.if.row_index === 'even' && realIndex % 2 === 0)
+                    ) &&
+                    (
+                        tt.if.filter === undefined ||
+                        (() => {
+                            const ast = new SyntaxTree(tt.if.filter);
+                            return ast.isValid && ast.evaluate(
+                                this.props.virtualized.data[realIndex - this.props.virtualized.offset.rows]
+                            );
+                        })()
+                    )
+                );
+        }, column_conditional_tooltips);
+
+        return conditionalTooltips.length ?
+            conditionalTooltips[0] :
+            legacyTooltip || staticTooltip;
+    }
+
+    componentDidUpdate() {
+        const { tooltip } = this.refs as { [key: string]: HTMLElement };
+        if (!tooltip) {
+            return;
+        }
+
+        // x-axis view start / end
+        const xvs = document.body.offsetLeft;
+        const xve = document.body.offsetLeft + document.body.clientWidth;
+
+        // x-asix tooltip start / end
+        const xts = tooltip.offsetLeft;
+        const xte = tooltip.offsetLeft + tooltip.clientWidth;
+
+        // x-axis view-tooltip delta start / end
+        // positive: tooltip is in the view
+        // negative: tooltip is out of view
+        // inverted to keep the same meaning on both sides
+        const xds = xts - xvs;
+        const xde = xve - xte;
+
+        if (xde >= 0) {
+            return;
+        }
+
+        // the tooltip can only exceed the view in one direction (right)
+        // bound the adjustment to whatever is smallest (the right excess, or the left space)
+        tooltip.style.marginLeft = `${Math.max(xde, -xds)}px`;
+    }
+
+    render() {
+        let children = (this.props as any).children;
+
+        const { displayTooltip } = this.state;
+
+        let tooltip = (displayTooltip || undefined) && this.getTooltip();
+
+        let tooltipComponent;
+        if (tooltip && !this.state.md) {
+            import(/* webpackChunkName: "markdown" */ 'remarkable')
```

Comment:
Dynamically load the markdown library. The component's state will be update when it becomes available to re-render itself correctly.

Reply:
neat - perhaps we should use this in `dcc` as well!

Reply:
Yes! There will be a few things to iron out like dependency packaging and reuse.. wouldn't do to do all this and reload your library multiple times b/c they come from different packages!